llvm/test/CodeGen/AIE/aie2/GlobalISel/legalize-fcmp-bf16.mir

```diff
@@ -0,0 +1,368 @@
+    ; CHECK-NEXT: [[AND:%[0-9]+]]:_(s32) = G_AND [[OR]], [[C1]]
+    ; CHECK-NEXT: $r0 = COPY [[AND]](s32)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $r0
+    %3:_(s32) = COPY $r1
+    %1:_(s16) = G_TRUNC %3:_(s32)
+    %4:_(s32) = COPY $r2
+    %2:_(s16) = G_TRUNC %4:_(s32)
+    %0:_(s1) = G_FCMP floatpred(one), %1:_(s16), %2:_
+    %5:_(s32) = G_ZEXT %0:_(s1)
+    $r0 = COPY %5:_(s32)
+    PseudoRET implicit $lr, implicit $r0
+...
+
+---
+name:            fcmp_oeq
+body:             |
+  bb.1.entry:
+    liveins: $r1, $r2
+    ; CHECK-LABEL: name: fcmp_oeq
+    ; CHECK: liveins: $r1, $r2
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s32) = COPY $r1
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(s32) = COPY $r2
+    ; CHECK-NEXT: [[DEF:%[0-9]+]]:_(<32 x s16>) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT1:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY1]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[INT:%[0-9]+]]:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vltbf16), [[AIE_INSERT_VECTOR_ELT]](<32 x s16>), [[AIE_INSERT_VECTOR_ELT1]](<32 x s16>)
+    ; CHECK-NEXT: [[INT1:%[0-9]+]]:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vltbf16), [[AIE_INSERT_VECTOR_ELT1]](<32 x s16>), [[AIE_INSERT_VECTOR_ELT]](<32 x s16>)
+    ; CHECK-NEXT: [[OR:%[0-9]+]]:_(s32) = G_OR [[INT]], [[INT1]]
+    ; CHECK-NEXT: [[C1:%[0-9]+]]:_(s32) = G_CONSTANT i32 -1
+    ; CHECK-NEXT: [[XOR:%[0-9]+]]:_(s32) = G_XOR [[OR]], [[C1]]
+    ; CHECK-NEXT: [[C2:%[0-9]+]]:_(s32) = G_CONSTANT i32 1
+    ; CHECK-NEXT: [[AND:%[0-9]+]]:_(s32) = G_AND [[XOR]], [[C2]]
+    ; CHECK-NEXT: $r0 = COPY [[AND]](s32)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $r0
+    %3:_(s32) = COPY $r1
+    %1:_(s16) = G_TRUNC %3:_(s32)
+    %4:_(s32) = COPY $r2
+    %2:_(s16) = G_TRUNC %4:_(s32)
+    %0:_(s1) = G_FCMP floatpred(oeq), %1:_(s16), %2:_
+    %5:_(s32) = G_ZEXT %0:_(s1)
+    $r0 = COPY %5:_(s32)
+    PseudoRET implicit $lr, implicit $r0
+...
+
+---
+name:            fcmp_ugt
+body:             |
+  bb.1.entry:
+    liveins: $r1, $r2
+    ; CHECK-LABEL: name: fcmp_ugt
+    ; CHECK: liveins: $r1, $r2
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s32) = COPY $r1
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(s32) = COPY $r2
+    ; CHECK-NEXT: [[DEF:%[0-9]+]]:_(<32 x s16>) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT1:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY1]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[INT:%[0-9]+]]:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vltbf16), [[AIE_INSERT_VECTOR_ELT1]](<32 x s16>), [[AIE_INSERT_VECTOR_ELT]](<32 x s16>)
```

Comment:
This doesn't check for NaNs. This can be handled by additional checking for `FCMP_ORD`:
```
ord: yields true if both operands are not a QNAN.
```